### PR TITLE
fix: domain resolution

### PR DESCRIPTION
**Pull request by hmalik88**
Fixes domain resolution results showing up on the send screen after inputs have been cleared.

Previous behavior: 

https://github.com/MetaMask/metamask-extension/assets/41640681/4d91fd7d-0405-4b36-ab06-c2c0e76649d6

Current behavior

https://github.com/MetaMask/metamask-extension/assets/41640681/bff9703b-f60c-4ab8-ba6f-6f4f596a8b28

### CODE REVIEW

**Review by GuillaumeRx — DISMISSED** [review]

**Review by FrederikBolding — CHANGES_REQUESTED** [review]
I can still reproduce the issue locally

File: ui/ducks/send/send.js [hunk 1]
```diff
@@ -1383,6 +1384,7 @@ const slice = createSlice({
             (!isValidHexAddress(state.recipientInput, {
               mixedCaseUseChecksum: true,
             }) &&
+              !IS_FLASK &&
```
Comment (FrederikBolding) [inline ui/ducks/send/send.js:1387]: Why is this needed?

Comment (hmalik88) [reply ui/ducks/send/send.js:1387]: `onChange` prop func in `DomainInput` calls `updateRecipientUserInput` which makes its way to this line of code and erroneously assigns an error if not hex in flask before we even get to the domain lookup call.

**Review by FrederikBolding — APPROVED** [review]

**Review by GuillaumeRx — APPROVED** [review]
